test/unittest/rtps/writer/CMakeLists.txt

```diff
@@ -93,19 +93,21 @@ add_gtest(LivelinessManagerTests SOURCES ${LIVELINESSMANAGERTESTS_SOURCE})
 
 set(RTPSWRITERTESTS_SOURCE RTPSWriterTests.cpp)
 
-add_executable(RTPSWriterTests ${RTPSWRITERTESTS_SOURCE})
-target_compile_definitions(RTPSWriterTests PRIVATE
-    BOOST_ASIO_STANDALONE
-    ASIO_STANDALONE
-    $<$<AND:$<NOT:$<BOOL:${WIN32}>>,$<STREQUAL:"${CMAKE_BUILD_TYPE}","Debug">>:__DEBUG>
-    $<$<BOOL:${INTERNAL_DEBUG}>:__INTERNALDEBUG> # Internal debug activated.
-    )
-target_include_directories(RTPSWriterTests PRIVATE
-    ${Asio_INCLUDE_DIR})
-target_link_libraries(RTPSWriterTests fastcdr fastrtps foonathan_memory
-    GTest::gmock
-    ${CMAKE_DL_LIBS})
-add_gtest(RTPSWriterTests SOURCES ${RTPSWRITERTESTS_SOURCE})
+if(NOT QNX)
```

Comment:
Just a small nitpick. Since we no longer need to build the RTPSWriterTests due to the WillOnce issue, better to move this before the `set(RTPSWRITERTESTS_SOURCE RTPSWriterTests.cpp)`